src/pool/srv_pool.c

```diff
@@ -6898,3 +6898,103 @@ ds_pool_svc_flush_map(struct ds_pool_svc *ds_svc, struct pool_map *map)
 out:
 	return rc;
 }
+
+int
+ds_pool_svc_update_label(struct ds_pool_svc *ds_svc, const char *label)
+{
+	struct pool_svc		*svc = pool_ds2svc(ds_svc);
+	daos_prop_t		*prop = NULL;
+	struct rdb_tx		 tx = { 0 };
+	int			 rc = 0;
+
+	prop = daos_prop_alloc(1);
+	if (prop == NULL)
+		D_GOTO(out, rc = -DER_NOMEM);
+
+	prop->dpp_entries[0].dpe_type = DAOS_PROP_PO_LABEL;
+	if (label != NULL) {
+		D_STRNDUP(prop->dpp_entries[0].dpe_str, label, strlen(label));
+		if (prop->dpp_entries[0].dpe_str == NULL)
+			D_GOTO(out, rc = -DER_NOMEM);
+	} else {
+		prop->dpp_entries[0].dpe_flags = DAOS_PROP_ENTRY_NOT_SET;
```

Comment:
It looks this case, pool_prop_write() will still write a special pool label "pool_label_not_set"

Reply:
Inside pool_prop_write(), for DAOS_PROP_PO_LABEL, if "entry->dpe_str == NULL", the pool label will be set as default as following:

static int
pool_prop_write(struct rdb_tx *tx, const rdb_path_t *kvs, daos_prop_t *prop)
{
...
                case DAOS_PROP_PO_LABEL:
                        if (entry->dpe_str == NULL ||
                            strlen(entry->dpe_str) == 0) {
                                entry = daos_prop_entry_get(&pool_prop_default,
                                                            entry->dpe_type);
                                D_ASSERT(entry != NULL);
                        }